### PR TITLE
[#156] Add local host volume

### PR DESCRIPTION
**Pull request by zipofar**

### CODE REVIEW

File: lib/uffizzi/services/compose_file_service.rb [hunk 1]
```diff
@@ -3,38 +3,91 @@
...
         }
       end
     rescue Errno::ENOENT => e
       dependency_path = e.message.split('- ').last
       raise Uffizzi::Error.new("The config file #{dependency_path} does not exist")
     end
 
-    def fetch_configs(configs_data)
+    def prepare_dependency_host_volumes_files(dependency_file_paths, compose_file_dir)
+      dependency_file_paths.map do |dependency_file_path|
```
Comment (artempartos) [inline lib/uffizzi/services/compose_file_service.rb:53]: can we simpify it and not use regexp?

something like
```
dependency_file_paths.map do |dependency_file_path|
  dependency_pathname = Pathname.new(dependency_file_path);
  next dependency_file_path if dependency_pathname.absolute?

  expended_path = dependency_pathname.expand_path(compose_file_dir)
  next expended_path.to_s if expended_path.exist?

  raise("Unsupported path #{dependency_file_path}")
end

```

File: lib/uffizzi/services/volume_parser_service.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      target_path = volume_data['target'].to_s.strip
+
+      source_path if host_volume?(source_path, target_path)
+    end
+
+    def host_volume?(source_path, target_path)
+      path?(source_path) && path?(target_path)
+    end
+
+    def path?(path)
```
Comment (artempartos) [inline lib/uffizzi/services/volume_parser_service.rb:41]: relative?

**Review by artempartos — CHANGES_REQUESTED** [review]

File: lib/uffizzi/services/compose_file_service.rb [hunk 3]
```diff
@@ -68,26 +121,14 @@ def parse_env_file(env_file)
...
-
-    def prepare_service_env_files(service_data)
-      env_files_data = []
-      service_data.each_pair do |key, value|
-        key_sym = key.to_sym
-        if key_sym == :env_file
-          env_files_data << parse_env_file(value)
+      services.map do |_, service_data|
+        service_data.map do |key, value|
+          parse_env_file(value) if key.to_sym == :env_file
```
Comment (moklidia) [inline lib/uffizzi/services/compose_file_service.rb:129]: Can we avoid nested map?

File: lib/uffizzi/services/volume_parser_service.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+      volumes.map do |volume|
+        case volume
+        when String
+          process_short_syntax(volume)
+        when Hash
+          process_long_syntax(volume)
+        else
+          Uffizzi.ui.say("Unsupported type of '#{volumes}' option")
+        end
+      end
```
Comment (moklidia) [inline lib/uffizzi/services/volume_parser_service.rb:19]: Move the block to a separate method maybe?

File: lib/uffizzi/services/volume_parser_service.rb [hunk 5]
```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+class VolumeParserService
+  class << self
+    def parse(volumes)
+      return [] if volumes.empty?
+
+      Uffizzi.ui.say("Volumes '#{volumes}' should be an array") if volumes.is_a?(String)
```
Comment (moklidia) [inline lib/uffizzi/services/volume_parser_service.rb:8]: How about? `unless volumes.is_a?(Array)`

File: lib/uffizzi/services/volume_parser_service.rb [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+      source_path if host_volume?(source_path, target_path)
+    end
+
+    def host_volume?(source_path, target_path)
+      path?(source_path) && path?(target_path)
+    end
+
+    def path?(path)
+      /^(\/|\.\/|\.\.\/)/.match?(path) # volume path should start with / or ./ or ../
```
Comment (moklidia) [inline lib/uffizzi/services/volume_parser_service.rb:44]: Should we refator it to not use the regex?

**Review by moklidia — APPROVED** [review]

**Pull request merged**